### PR TITLE
Eliminate forbidden name restriction for OmnisciDB 5.2.

### PR DESCRIPTION
**Pull request by pearu**
The forbidden name restriction is removed by using name mangling `__<id>` for all low-level function names. This requires the usage of https://github.com/omnisci/omniscidb-internal/pull/4133

This PR also revised omnisci math tests and fixes various bugs with boolean support.

Fixes: #32

### CODE REVIEW

File: rbc/typesystem.py [hunk 1]
```diff
@@ -1088,7 +1092,15 @@ def get_data_type(self):
...
-        pass
+
+        def can_convert_to(self, typingctx, other):
+            return isinstance(other, numba.types.Boolean)
+
+        def can_convert_from(self, typingctx, other):
+            return isinstance(other, numba.types.Boolean)
+
+        def unify(self, context, other):
+            print('UNIFY', other)
```
Comment (guilhermeleobas) [inline rbc/typesystem.py]: Are `unify` missing its implementation?

File: rbc/typesystem.py [hunk 2]
```diff
@@ -1104,6 +1116,16 @@ def get_value_type(self):
...
 
+    @numba.targets.imputils.lower_cast(numba.types.Boolean, Boolean1)
+    @numba.targets.imputils.lower_cast(numba.types.Boolean, Boolean8)
+    def literal_boolean_to_booleanN(context, builder, fromty, toty, val):
+        llty = context.get_value_type(toty)
+        return builder.zext(val, llty)
+
+    @numba.extending.unbox(Boolean8)
+    def unbox_boolean8(typ, obj, c):
+        print('unbox_boolean8', typ, obj)
```
Comment (guilhermeleobas) [inline rbc/typesystem.py]: missing implementation?

File: rbc/typesystem.py [hunk 3]
```diff
@@ -1120,3 +1142,73 @@ def model__init__(self, dmm, fe_type):
...
+    if np is not None:
+        if isinstance(obj, np.ufunc):
+            parameters = dict()
+            returns = dict()
+
+            sigline = obj.__doc__.lstrip().splitlines(1)[0]
+            m = _ufunc_pos_args_match(sigline)
+            name = m['name']
+            assert name == obj.__name__, (name, obj.__name__)
+            # rest = m['rest']
```
Comment (guilhermeleobas) [inline rbc/typesystem.py]: remove this comment

File: rbc/typesystem.py (hunk 1 above)
Comment (pearu) [reply rbc/typesystem.py]: Removed.

File: rbc/typesystem.py (hunk 2 above)
Comment (pearu) [reply rbc/typesystem.py]: Removed as not needed atm.

File: rbc/typesystem.py (hunk 3 above)
Comment (pearu) [reply rbc/typesystem.py]: done

**Pull request merged**